src/app/modules/core/services/beacon-node.service.ts

```diff
@@ -25,16 +41,58 @@ export class BeaconNodeService {
   ) { }
 
   private apiUrl = this.environmenter.env.validatorEndpoint;
-  private beaconNodeEndpoint: string;
-
-  // Observables.
-  status$ = this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).pipe(
-    tap((res: NodeConnectionResponse) => {
-      this.beaconNodeEndpoint = res.beaconNodeEndpoint + BEACON_API_SUFFIX;
-    }),
-  );
+  beaconNodeState: BeaconNodeState = new BeaconNodeState();
+
+  private get getBeaconEndpoint$(): BehaviorSubject<string> {
+    if (!this.beaconNodeState.beaconNodeEndpoint$.getValue()) {
+      this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).subscribe(
+          (res: NodeConnectionResponse) => {
+          this.beaconNodeState.beaconNodeEndpoint$.next("http://" + res.beaconNodeEndpoint + BEACON_API_SUFFIX);
+          },
+          (error) => {
+          }
+        );
+    }
+    return this.beaconNodeState.beaconNodeEndpoint$;
+  }
+
+  private updateBalances(): void {
+    this.getBeaconEndpoint$.pipe(
+      filter((beacanNodeEnpoint) => {
+        return beacanNodeEnpoint !== undefined;
+      }),
+      map(beacanNodeEnpoint => {
+        return this.http.get<ValidatorBalances>(`${beacanNodeEnpoint}/validators/balances`)
+          .subscribe((result) => {
+            this.beaconNodeState.balances$.next(result)
+            this.beaconNodeState.connected$.next(true);
+          },
+          (err) => {
+            this.beaconNodeState.connected$.next(false);
+          });
+      })
+    ).subscribe();
+  }
+
+  private updateSyncing(): void {
```

Comment:
Where is this called? Wouldn't an observable polling pattern work best here?

```js
status$ = this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`);
statusPoll$ = interval(POLLING_INTERVAL).pipe(
    startWith(0),
    mergeMap(_ => this.status$),
);
```
Where `statusPoll$` is used by the beacon-node-status component to continuously display changes in node status, such as when the beacon node disconnects, or when it finishes syncing, etc.